ci-operator/step-registry/upi/install/gcp/upi-install-gcp-commands.sh

```diff
@@ -102,9 +109,14 @@ if [[ -s "${SHARED_DIR}/xpn.json" ]]; then
   HOST_PROJECT_COMPUTE_SERVICE_ACCOUNT="$(jq -r '.computeServiceAccount' "${SHARED_DIR}/xpn.json")"
   HOST_PROJECT_CONTROL_SERVICE_ACCOUNT="$(jq -r '.controlServiceAccount' "${SHARED_DIR}/xpn.json")"
   HOST_PROJECT_PRIVATE_ZONE_NAME="$(jq -r '.privateZoneName' "${SHARED_DIR}/xpn.json")"
-else
-  # Set HOST_PROJECT to the cluster project so commands with `--project` work in both scenarios.
-  HOST_PROJECT="${PROJECT_NAME}"
+  PROJECT_OPTION="--project ${HOST_PROJECT}"
+
+  if [[ -f "${CLUSTER_PROFILE_DIR}/xpn_project_setting.json" ]]; then
```

Comment:
The content of `"${CLUSTER_PROFILE_DIR}/xpn_project_setting.json"` should be the same as `"${SHARED_DIR}/xpn.json"`, sounds like we can figure out a way to remove this if condition.